app/models/protected_area.rb

```diff
@@ -9,6 +9,7 @@ class ProtectedArea < ActiveRecord::Base
   has_many :networks_protected_areas, dependent: :destroy
   has_many :networks, through: :networks_protected_areas
   has_many :pame_evaluations
+  has_many :story_map_links, primary_key: :wdpa_id, foreign_key: :wdpa_id
```

Comment:
You don't need the `primary_key` definition here.
The `foreing_key` is the one that will handle the relationship

Reply:
But to be honest I think you can remove the foreign_key as well and use the default id to handle the relationship.
Even though the links are related to the wdpa_id, here we are working with database objects which have also a unique id for each wdpa_id. So specifying a different foreign_key it's probably unnecessary.

You can also check how other relationships are made within PP